### DISCUSSION TITLE
[Checkbox] use standard values for checked attribute

### DISCUSSION THREAD

**Discussion by rtritto** [saadeghi/daisyui#3150] [Ideas / feature requests]
For `checked` attribute, instead of value (`"checked" | undefined`):
```html
<input type="checkbox" checked="checked" class="checkbox" />
---
<input type="checkbox" class="checkbox" />
```

use the standard values (`boolean | undefined`):
```html
<input type="checkbox" checked class="checkbox" />
---
<input type="checkbox" checked={false} class="checkbox" />
OR
<input type="checkbox" class="checkbox" />
```

**Comment by saadeghi** [comment]
https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attributes

>The values "true" and "false" are not allowed on boolean attributes.


>Here is an example of a checkbox that is checked and disabled. The [checked](https://html.spec.whatwg.org/multipage/input.html#attr-input-checked) and [disabled](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#attr-fe-disabled) attributes are the boolean attributes.
>```
><label><input type=checkbox checked name=cheese disabled> Cheese</label>
>```
>This could be equivalently written as this:
>```
><label><input type=checkbox checked=checked name=cheese disabled=disabled> Cheese</label>
>```
>You can also mix styles; the following is still equivalent:
>```
><label><input type='checkbox' checked name=cheese disabled=""> Cheese</label>
>```

**Reply by rtritto** [reply]
I was confused in JSX with:
```jsx
<input type="checkbox" checked="checked" class="checkbox" />
```
because I got:
```console
Type 'string' is not assignable to type 'boolean | undefined'.ts(2322)
jsx.d.ts(911, 5): The expected type comes from property 'checked' which is declared here on type 'InputHTMLAttributes<HTMLInputElement>'
(property) JSX.InputHTMLAttributes<HTMLInputElement>.checked?: boolean | undefined
```

Anyway I resolved using boolean value, thanks!
